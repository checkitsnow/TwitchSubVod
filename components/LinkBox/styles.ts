import styled from 'styled-components';

export const Container = styled.div`
  display: flex;
  flex-direction: column;
  margin-top: 1rem;

  span {
    display: flex;
    flex-direction: row;
    align-items: center;
    cursor: pointer;

    svg {
      margin-right: 0.5rem;
    }
    a {
      color: var(--text);
    }
  }
`;
