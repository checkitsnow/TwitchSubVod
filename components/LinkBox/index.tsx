import React from 'react';
import { FiSearch } from 'react-icons/fi';
import Link from 'next/link';

import { Container } from './styles';

interface LinksProps {
  home?: boolean;
  clips?: boolean;
  vods?: boolean;
  download?: boolean;
}

const LinkBox = ({ home, clips, vods, download }: LinksProps) => {
  return (
    <Container>
      {home && (
        <Link href="/" aria-label="Home">
          <span>
            <FiSearch size={14} />
            Home
          </span>
        </Link>
      )}

      {clips && (
        <Link href="/DeletedClips" aria-label="DeletedClips">
          <span>
            <FiSearch size={14} />
            Search for deleted clips
          </span>
        </Link>
      )}

      {vods && (
        <Link href="/DeletedVods" aria-label="DeletedVods">
          <span>
            <FiSearch size={14} />
            Search for deleted vods
          </span>
        </Link>
      )}

      {download && (
        <Link href="/DownloadClip" aria-label="DownloadClip">
          <span>
            <FiSearch size={14} />
            Download Twitch Clips
          </span>
        </Link>
      )}
    </Container>
  );
};

export default LinkBox;
