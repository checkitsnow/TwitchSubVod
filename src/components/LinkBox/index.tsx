import React from 'react';
import { FiSearch } from 'react-icons/fi';
import { Link } from 'react-router-dom';

import { Container } from './styles';

interface LinksProps {
  home?: boolean;
  clips?: boolean;
  vods?: boolean;
  download?: boolean;
}

const LinkBox = ({ home, clips, vods, download }: LinksProps) => {
  return (
    <Container>
      {home && (
        <Link to="/">
          <FiSearch size={14} />
          Home
        </Link>
      )}

      {clips && (
        <Link to="/DeletedClips">
          <FiSearch size={14} />
          Seach for deleted clips
        </Link>
      )}

      {vods && (
        <Link to="/DeletedVods">
          <FiSearch size={14} />
          Seach for deleted vods
        </Link>
      )}

      {download && (
        <Link to="/DownloadClip">
          <FiSearch size={14} />
          Download Twitch Clips
        </Link>
      )}
    </Container>
  );
};

export default LinkBox;
