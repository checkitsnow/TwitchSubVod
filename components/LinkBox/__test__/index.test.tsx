import { render } from '@testing-library/react';
import LinkBox from '..';

it('should render LinkBoxes correctly', () => {
  const tree = render(
    <>
      <LinkBox home />
      <LinkBox clips />
      <LinkBox vods />
      <LinkBox download />
      <LinkBox all />
    </>,
  );

  expect(tree).toMatchSnapshot();
});
