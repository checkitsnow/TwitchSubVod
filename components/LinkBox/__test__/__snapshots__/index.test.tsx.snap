// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LinkBoxes correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Container-cfnpnh-0 fkFsiv"
      >
        <span>
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          Home
        </span>
      </div>
      <div
        class="styles__Container-cfnpnh-0 fkFsiv"
      >
        <span>
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          Search for deleted clips
        </span>
      </div>
      <div
        class="styles__Container-cfnpnh-0 fkFsiv"
      >
        <span>
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          Search for deleted vods
        </span>
      </div>
      <div
        class="styles__Container-cfnpnh-0 fkFsiv"
      >
        <span>
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <circle
              cx="11"
              cy="11"
              r="8"
            />
            <line
              x1="21"
              x2="16.65"
              y1="21"
              y2="16.65"
            />
          </svg>
          Download Twitch Clips
        </span>
      </div>
      <div
        class="styles__Container-cfnpnh-0 fkFsiv"
      >
        <span>
          Home
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </span>
        <span>
          Search for Deleted Clips
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </span>
        <span>
          Search for Deleted Vods
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </span>
        <span>
          Download Twitch Clips
          <svg
            fill="none"
            height="14"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="14"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
            />
            <polyline
              points="15 3 21 3 21 9"
            />
            <line
              x1="10"
              x2="21"
              y1="14"
              y2="3"
            />
          </svg>
        </span>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Container-cfnpnh-0 fkFsiv"
    >
      <span>
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        Home
      </span>
    </div>
    <div
      class="styles__Container-cfnpnh-0 fkFsiv"
    >
      <span>
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        Search for deleted clips
      </span>
    </div>
    <div
      class="styles__Container-cfnpnh-0 fkFsiv"
    >
      <span>
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        Search for deleted vods
      </span>
    </div>
    <div
      class="styles__Container-cfnpnh-0 fkFsiv"
    >
      <span>
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <circle
            cx="11"
            cy="11"
            r="8"
          />
          <line
            x1="21"
            x2="16.65"
            y1="21"
            y2="16.65"
          />
        </svg>
        Download Twitch Clips
      </span>
    </div>
    <div
      class="styles__Container-cfnpnh-0 fkFsiv"
    >
      <span>
        Home
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </span>
      <span>
        Search for Deleted Clips
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </span>
      <span>
        Search for Deleted Vods
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </span>
      <span>
        Download Twitch Clips
        <svg
          fill="none"
          height="14"
          stroke="currentColor"
          stroke-linecap="round"
          stroke-linejoin="round"
          stroke-width="2"
          viewBox="0 0 24 24"
          width="14"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M18 13v6a2 2 0 0 1-2 2H5a2 2 0 0 1-2-2V8a2 2 0 0 1 2-2h6"
          />
          <polyline
            points="15 3 21 3 21 9"
          />
          <line
            x1="10"
            x2="21"
            y1="14"
            y2="3"
          />
        </svg>
      </span>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
